core/services/configuration_generator.rb

```diff
@@ -1,23 +1,25 @@
 # frozen_string_literal: true
 
 # The class provides methods for generating the role of the file.
-class ConfigurationGenerator
-  # Generate a list of role parameters in JSON format
+module ConfigurationGenerator
   # @param box_definitions [BoxDefinitions] the list of BoxDefinitions that are configured in the application
   # @param name [String] node name
-  # @param product_config [Hash] list of the product parameters
-  # @param recipe_name [String] name of the recipe
+  # @param product_configs [Hash] list of the product parameters
+  # @param recipes_names [String] name of the recipe
   # @param box [String] name of the box
   # @param rhel_credentials redentials for subscription manager
-  def self.generate_json_format(box_definitions, name, product_configs, recipes_names, box, rhel_credentials)
+  def self.generate_json_format(name, recipes_names, product_configs = {},
+                                box = nil, box_definitions = nil, rhel_credentials = nil)
     run_list = ['recipe[mdbci_provision_mark::remove_mark]',
                 *recipes_names.map { |recipe_name| "recipe[#{recipe_name}]" },
                 'recipe[mdbci_provision_mark::default]']
-    if check_subscription_manager(box_definitions, box)
-      raise 'RHEL credentials for Red Hat Subscription-Manager are not configured' if rhel_credentials.nil?
+    unless box_definitions.nil?
+      if check_subscription_manager(box_definitions, box)
+        raise 'RHEL credentials for Red Hat Subscription-Manager are not configured' if rhel_credentials.nil?
 
-      run_list.insert(1, 'recipe[subscription-manager]')
-      product_configs = product_configs.merge('subscription-manager': rhel_credentials)
+        run_list.insert(1, 'recipe[subscription-manager]')
+        product_configs = product_configs.merge('subscription-manager': rhel_credentials)
```

Comment:
This approach is wrong. The merge command  will create a new hash that will contain data both from the product_configs hash and method's properties. In the result, the source hash will not contain the required properties.